cmd/issueZKCert.go

```diff
@@ -108,6 +108,11 @@ func issueZKCert(f *issueZKCertFlags) error {
 		return fmt.Errorf("connect to blockchain rpc: %w", err)
 	}
 
+	chainID, err := client.ChainID(ctx)
+	if err != nil {
+		return fmt.Errorf("could not retrieve chain-id: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("retrieve chain-id: %w", err)
```

Rule: don't write "failed"/"cannot"/"error" in error message — it's already an error, those extra words create unnecessary noise in error messages; instead write the action that what you were trying to do (in bare infinitive form).

Reply:
It's not yet a common practice, but from our experience this is the best signal-to-noise ratio way to do it. There are opinions in the community for and against this approach.

Unfortunately, as we don't have a dedicated knowledge base, we don't have it written anywhere :(